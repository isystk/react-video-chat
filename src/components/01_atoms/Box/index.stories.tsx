import { Meta, Story } from '@storybook/react'
import React from 'react'
import Box from './index'

export default {
  title: '01_atoms/Box',
  component: Box,
} as Meta

const Template: Story = (props) => {
  return <Box title="sample" />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
