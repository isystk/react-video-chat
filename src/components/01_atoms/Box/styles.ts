import { css, keyframes } from '@emotion/css'

export const breadcrumbs = css`
    padding: 20px 0px;
`

export const box = css`
    background-color: #fff;
    height: 80vh;
    position: relative;
    text-align: center;
`

export const title = css`
    background-color: #eee;
    padding: 10px 20px;
    text-align: left;
`
