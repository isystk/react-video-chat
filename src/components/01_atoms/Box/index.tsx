import React, { FC } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { Box, Breadcrumbs, Typography } from '@material-ui/core'
import { connect } from '@/components/hoc'
import { useStyles } from './styles'
import Link from 'next/link'
import { Url } from '@/constants/url'

/** BoxProps Props */
export type BoxProps = WithChildren & { title }
/** Presenter Props */
export type PresenterProps = BoxProps & { classes }

/** Presenter Component */
const BoxPresenter: FC<PresenterProps> = ({
  children,
  title,
  classes,
  ...props
}) => (
  <>
    <Breadcrumbs aria-label="breadcrumb" className={classes.breadcrumbs}>
      <Link color="inherit" href={Url.TOP}>
        TOP
      </Link>
      <Typography component="span">{title}</Typography>
    </Breadcrumbs>
    <Box component="div" className={classes.box}>
      <Typography component="div" color="primary" className={classes.title}>
        {title}
      </Typography>
      {children}
    </Box>
  </>
)

/** Container Component */
const BoxContainer: React.FC<ContainerProps<BoxProps, PresenterProps>> = ({
  presenter,
  children,
  ...props
}) => {
  const classes = useStyles()
  return presenter({
    children,
    classes,
    ...props,
  })
}

export default connect<BoxProps, PresenterProps>(
  'Box',
  BoxPresenter,
  BoxContainer
)
