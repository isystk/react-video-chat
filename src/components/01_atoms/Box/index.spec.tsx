import React from 'react'
import renderer from 'react-test-renderer'
import Box from './index'
import '@testing-library/jest-dom/extend-expect'

describe('Box', () => {
  it('Match Snapshot', () => {
    const component = renderer.create(<Box title="sample" />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
