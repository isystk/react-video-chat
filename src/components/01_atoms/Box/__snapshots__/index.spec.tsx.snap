// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Match Snapshot 1`] = `
Array [
  <nav
    aria-label="breadcrumb"
    className="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1kynpjt css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      className="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
    >
      <li
        className="MuiBreadcrumbs-li"
      >
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          TOP
        </a>
      </li>
      <li
        aria-hidden={true}
        className="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        className="MuiBreadcrumbs-li"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          sample
        </span>
      </li>
    </ol>
  </nav>,
  <div
    className="css-heco0o MuiBox-root css-0"
  >
    <div
      className="MuiTypography-root MuiTypography-body1 css-13emhng css-vkc4v8-MuiTypography-root"
    >
      sample
    </div>
  </div>,
]
`;
